v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go

```diff
@@ -263,5 +267,5 @@ func (c *armConversionApplier) createExtensionResourceOwnerProperty() *astmodel.
 	return astmodel.NewPropertyDefinition(
 		c.idFactory.CreatePropertyName(astmodel.OwnerProperty, astmodel.Exported),
 		c.idFactory.CreateIdentifier(astmodel.OwnerProperty, astmodel.NotExported),
-		astmodel.ArbitraryOwnerReference).MakeRequired()
+		astmodel.NewOptionalType(astmodel.ArbitraryOwnerReference)).MakeRequired()
```

Comment:
Should "MakeRequired()" enforce the optional wrapper?

Reply:
I don't want to have `MakeRequired` return an `err`, but agree this is a good idea. I think it makes sense to make `MakeRequired` panic in this case, as really the problem is a programming error (you have asked the property to move into a nonsensical state).

Thoughts?

Reply:
Oh, I was thinking that `MakeRequired()` could proactively wrap the property type if it's not already optional, so the call site would simplify:

```suggestion
		astmodel.ArbitraryOwnerReference).MakeRequired()
```

Reply:
Hmm, we could but it feels a bit strange to have `MakeRequired` actually do `MakeTypeOptional` internally? It would also make `MakeRequired` and `MakeOptional` non-symmetric as one would internally do `MakeTypeOptional` but I don't think that `MakeOptional` could do `MakeTypeRequired`, which feels weird?

Reply:
Fair enough.